client/layout/masterbar/publish.jsx

```diff
@@ -75,17 +75,9 @@ class MasterbarItemNew extends React.Component {
 		return 'bottom left';
 	}
 
-	onSiteSelect = siteId => {
-		this.props.recordTracksEvent( 'calypso_masterbar_write_button_clicked' );
-		//To avoid binding in connect, please remove me later.
-		const redirectURL = getEditorUrl( reduxGetState(), siteId, null, 'post' );
-		if ( typeof window !== 'undefined' ) {
-			setTimeout( () => {
-				window.location = redirectURL;
-			}, 0 );
-			return true; // handledByHost = true, don't let the component nav
-		}
-		return false; //otherwise fallback to normal handling
+	onSiteSelect = () => {
+		this.props.openEditor( this.props.editorUrl );
+		return true; // handledByHost = true, don't let the component nav
```

Comment:
Thanks for the amazing comment!!!

Reply:
Don't thank me, I've just copied whatever @gwwar wrote in the previous iteration 😏